### PR TITLE
`FormView` - Implement UI test cases 3.4, 3.5, 3.6

### PR DESCRIPTION
**Pull request by dfeinzimer**

### CODE REVIEW

**Comment by dfeinzimer** [comment]
This is nearly ready, theres is one new outstanding issue in the test design PR.

File: Sources/ArcGISToolkit/Components/Form/FormInputs/ComboBoxInput.swift [hunk 1]
```diff
@@ -107,13 +107,15 @@ struct ComboBoxInput: View {
...
                     .accessibilityIdentifier("\(element.label) Value")
-                if selectedValue == nil {
-                    Image(systemName: "list.bullet")
-                        .foregroundColor(.secondary)
-                        .accessibilityIdentifier("\(element.label) Options Button")
-                } else if !inputModel.isRequired && inputModel.isEditable  {
-                    ClearButton { selectedValue = nil }
-                        .accessibilityIdentifier("\(element.label) Clear Button")
+                if inputModel.isEditable {
+                    if selectedValue == nil {
```
Comment (mhdostal) [inline Sources/ArcGISToolkit/Components/Form/FormInputs/ComboBoxInput.swift:111]: What if the field does not support `nil` values or the value of `ComboBoxFormInput.noValueOption == .hide`? We wouldn't want to show the `Clear` button in that case, but probably the `list.bullet` image.

**Review by mhdostal — COMMENTED** [review]
Just the one comment.

File: Sources/ArcGISToolkit/Components/Form/FormInputs/ComboBoxInput.swift (hunk 1 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/Form/FormInputs/ComboBoxInput.swift:111]: I had thought so too but Android is placing a clear button in required fields. See [this conversation](https://devtopia.esri.com/runtime/common-toolkit/pull/224#discussion_r1065234) in the test design PR.

Kaushik highlighted that "After Clear" in the design spec implies that required fields can be cleared out:
Screenshot 2023-10-12 at 09 58 56

and Field Maps also places a clear button on a required combo field:
IMG_0041

As for `ComboBoxFormInput.noValueOption == .hide` the spec is:
Screenshot 2023-10-12 at 10 01 08 which is how we have [implemented](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/77f4e5ae12ac8ff24d3c4b7e85105df17f99ccfe/Sources/ArcGISToolkit/Components/Form/FormInputs/ComboBoxInput.swift#L230-L232) it.

**Review by mhdostal — APPROVED** [review]

**Pull request merged**